trie/hasher.go

```diff
@@ -181,7 +181,8 @@ func (h *hasher) encodeFullNode(n *fullNode) []byte {
 // This convention exists because node.encode can only be inlined/escape-analyzed when
 // called on a concrete receiver type.
 func (h *hasher) encodedBytes() []byte {
-	h.tmp = h.encbuf.AppendToBytes(h.tmp[:0])
```

Comment:
In this case, h.tmp[:0] always has the capacity for encoding node, i think no allocation will occur.

Reply:
Oh no it will.

Reply:
The test code:
```
func TestCC(t *testing.T) {
	appendToBytes := func(dst []byte) []byte {
		var size = 4
		out := append(dst, make([]byte, size)...)
		copy(out[len(dst):], []byte{4, 5, 6, 7})
		return out
	}

	tmp := make([]byte, 10)
	t.Logf("slice point: %p, cap: %d, data: %v", &tmp[0], cap(tmp), tmp)
	tmp = appendToBytes(tmp[:0])
	t.Logf("slice point: %p, cap: %d, data: %v", &tmp[0], cap(tmp), tmp)
}
```

The result:
```
=== RUN   TestCC
    encode_test.go:650: slice point: 0x14000104310, cap: 10, data: [0 0 0 0 0 0 0 0 0 0]
    encode_test.go:652: slice point: 0x14000104310, cap: 10, data: [4 5 6 7]
--- PASS: TestCC (0.00s)
```